packages/framework/package.json

```diff
@@ -211,7 +211,6 @@
   "devDependencies": {
     "@apidevtools/json-schema-ref-parser": "11.6.4",
     "@arethetypeswrong/cli": "^0.16.4",
-    "@nestjs/common": "10.4.1",
```

Comment:
<!-- Issues highlight specific problems with the subject under review. These problems can be user-facing or behind the scenes. It is strongly recommended to pair this comment with a suggestion. If you are not sure if a problem exists or not, consider leaving a question.  -->
⚠ **issue:** We need to keep this so that the test suites have the needed types. Please revert the removal.‏

Reply:
That's strange, as `@novu/api` declares `@nestjs/common` as a `dependency` [here](https://github.com/novuhq/novu/pull/6754/files#diff-2c40985d6d91eed8ae85ec1c8e754a85984ee32e156a600d2b7a467423d7e338R36) which means it should be installed when using the `--prod` flag.

Could you please share the error log for further debugging?

I believe reverting the removal of this dependency is a blocker for Framework tests.

Reply:
Hey @rifont, Please find error below:
<img width="1236" alt="Screenshot 2024-10-29 at 11 49 50 AM" src="https://github.com/user-attachments/assets/b52ae680-c5ee-416f-83cf-75083236e081">

Reply:
This is now solved. See https://github.com/novuhq/novu/pull/6754/commits/4a3e98d6be620f52a25aec47a057fdd4c2033e58. 

The trick is that @novu/framework requires a range for @nestjs/common peer dependency but by pinning the @nestjs/common version via PNPM overrides we effectively teach PNPM to add the correct dependencies when installing only production dependencies.

Reply:
I tested locally as well. It worked fabulous for me. Sok, we need an article for all of this 😄 .

Reply:
We will! 😆